### PR TITLE
C_CPP : Introduction to Friend function with examples 

### PR DESCRIPTION
**Pull request by shreya0204**
<hr>

## Description 📜

Added introduction to friend function along with examples and explanation in documentation format.
Fixes (https://github.com/girlscript/winter-of-contributing/issues/883)

<hr>

## Type of change 📝

- [x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x] C/CPP

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

File: Frontend_Web_Development_React_Angular_Vue/React/Week 1/All_About_npm.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+</br>
+Create a JavaScript file inside this folder using the touch command.
+
+<p><img src="assets/../../../assets/How_To_Use_NPM1.png"></p>
+
+Now use this command for initialising npm 
+```
+npm init
+```
+After this you will get a number of entries to fill and you can fill them according to your preference. All this will lead to the creation of a file with name `package` in your main folder. Once you'll start installing external packages they'll also show up in this package file.
```
Comment (rohan-kulkarni-25) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 1/All_About_npm.md:127]: - Please replace `package` by `package.json`

File: Frontend_Web_Development_React_Angular_Vue/React/Week 1/All_About_npm.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+[Here](https://www.npmjs.com/) you can search your desired packages.
```
Comment (rohan-kulkarni-25) [inline Frontend_Web_Development_React_Angular_Vue/React/Week 1/All_About_npm.md:129]: - You can replace this link by `On [npm](https://www.npmjs.com/) you can search your desired packages.`

**Review by rohan-kulkarni-25 — CHANGES_REQUESTED** [review]

**Comment by shreya0204** [comment]
@Saviour1001 Sorry for tagging but please review this PR. I have followed all the step-by-step procedures you have said in the meeting but I don't know why I am seeing other's commits and other's files changed in mine. Can you please see this and tell me what I made wrong 🙁

**Comment by Suvraneel** [comment]
@shreya dont worry about the changed files by others, we'll see only the last 2 made by you

**Comment by shreya0204** [comment]
> @shreya0204 dont worry about the changed files by others, we'll see only the last 2 made by you

Thanks for the confirmation, I was worried thinking about what I made wrong :)

**Comment by Suvraneel** [comment]
@shreya0204 Since that time much has changed in the repo

Do this in your PR branch:  Fetch > Compare > pull from girlscript/C_CPP to your main branch

image

**Comment by shreya0204** [comment]
> @shreya0204 Acually your worries were correct...
> 
> Do this in your PR branch: Fetch upstream > Compare > pull from girlscript/C_CPP to your friend-function branch
> 
> image

Yes, I have fetched and merged in my branch do I have to make a new PR?

**Comment by Suvraneel** [comment]
@shreya0204 Not from girlscript:main 🤦🏻 
From girlscript/C_CPP branch

Do this in your PR branch: Fetch upstream > Compare > pull from girlscript/C_CPP to your friend-function branch

**Comment by shreya0204** [comment]
@Suvraneel Sorry 😅Yeah It's done now, I have merged from C_CPP branch to friend-function branch

**Review by Suvraneel — CHANGES_REQUESTED** [review]
Content is great, passed plagiarism checks too...

one last thing, before merge -
1. Move this file to C_CPP/Virtual Functions, Friend Functions & Friend Classes/
2. Add a line for this in C_CPP/Virtual Functions, Friend Functions & Friend Classes/readme.md

**Comment by shreya0204** [comment]
@Suvraneel Thanks. It's done 👍

**Comment by Suvraneel** [comment]
@shreya0204 GG ❤️ :octocat: 💯

**Comment by shreya0204** [comment]
@Suvraneel Yay yay! Thank you so much!

**Pull request merged**